src/provider/model/ContextChannel.ts

```diff
@@ -86,3 +86,38 @@ export class SystemContextChannel extends ContextChannelBase {
         };
     }
 }
+
+export class AppContextChannel extends ContextChannelBase {
+    public readonly type!: 'app';
+    public readonly name: string;
+
+    private _context: Context | null;
+
+    public constructor(id: ChannelId, name: string) {
+        super(id, 'app');
+
+        this.name = name;
+
+        this._context = null;
+    }
+
+    public getStoredContext(): Context | null {
+        return this._context;
+    }
+
+    public setLastBroadcastContext(context: Context) {
```

Comment:
`setLastBroadcastContext`, `clearStoredContext`, `getStoredContext` methods and `_context ` feel like they will be relevant to (and identical in) every subclass extension apart from `Default` from this point on. Is this a fair assumption?

Would it be cleaner to define all these behaviour in `ContextChannelBase` and make `DefaultContextChannel` a special case where you override the behaviour?

Reply:
Not keen on overrides - (having to reason that the thing you're stopping being invoked is ok not to invoke, and in this case there'd be unused state on the default channel) but have moved shared behaviour into a base class.